src/main/java/ch/njol/skript/effects/EffPotion.java

```diff
@@ -132,15 +134,18 @@ protected void execute(final Event e) {
 		for (final LivingEntity en : entities.getArray(e)) {
 			for (final PotionEffectType t : ts) {
 				int duration = d;
-				if (en.hasPotionEffect(t)) {
-					for (final PotionEffect eff : en.getActivePotionEffects()) {
-						if (eff.getType() == t) {
-							duration += eff.getDuration();
-							break;
+				if (replaceExisting) {
+					en.addPotionEffect(new PotionEffect(t, duration, a, ambient, particles), true);
+				} else {
+					if (en.hasPotionEffect(t)) {
+						for (final PotionEffect eff : en.getActivePotionEffects()) {
+							if (eff.getType() == t) {
+								duration += eff.getDuration();
+								break;
+							}
 						}
 					}
 				}
-				en.addPotionEffect(new PotionEffect(t, duration, a, ambient, particles), true);

```

Comment:
Doesn't this prevent applying new effects altogether without using replace syntax?

Reply:
Looks like I missed this. It will not work it if replace syntax is not used and the user does not currently have that potion effect, so I will fix this soon.